ray-operator/apis/ray/v1/rayjob_types.go

```diff
@@ -111,6 +112,10 @@ type RayJobSpec struct {
 	EntrypointResources string `json:"entrypointResources,omitempty"`
 	// Configurations of submitter k8s job.
 	SubmitterConfig *SubmitterConfig `json:"submitterConfig,omitempty"`
+	// Specifies the number of retries before marking this job failed.
+	// Each retry creates a new RayCluster.
+	// +kubebuilder:default:=0
+	BackoffLimit *int32 `json:"backoffLimit,omitempty"`
```

Comment:
Out of curiosity, what's the reason to use `*int32` instead of `int32`? This PR checks whether it is nil or not multiple times.

Reply:
I was being safe in case a user updates the kuberay-operator version before they update the CRD manifest. In that case all the code for backoffLimit is guarded by `rayJobInstance.Spec.BackoffLimit != nil` and never runs. Also, it effects the omitempty behavior, if it's not a pointer we will omit the field if set to 0.

Reply:
Thinking about it more, this maybe doesn't need to be a pointer. Unlike with `numOfHosts`, if we default backOffLimit to 0 there should be no changes in behavior if there's version skew between kuberay-operator and the CRD. So I guess this mainly boils down to the omitempty behavior. This would need to be a pointer if we defaulted backoffLimit to a value > 1 though.

Reply:
@kevin85421 I'm keeping this as a pointer per the recommendation in the Kubernetes API conventions for optoinal / omitempty fields: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required